### PR TITLE
chore(api): add descriptions and notes to Accordion components

### PR DESCRIPTION
**Pull request by smritimadan**
This PR aims to restore and improve component documentation while following our content standards. Consideration was given to:

- providing information in a consistent, simple, clear and accessible way
- avoiding overloading with text or repetition
- using ways to make the content easier to scan quickly

## Checklist

- [x] :ok_hand: website updates are Garden Designer approved (add the designer as a reviewer)
- [x] :black_nib: copy updates are approved (add the content strategist as a reviewer)
- [x] :link: considered opportunities for adding cross-reference URLs (grep for keywords)
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: src/pages/components/timeline.mdx [hunk 1]
```diff
@@ -136,16 +148,25 @@ import COMPONENTS from '../../examples/accordions/components.ts';
 
 <Component components={COMPONENTS} componentName="timeline.content" />
 
+The Content component renders the provided content with a separator and an icon for the
```
Comment (jzempel) [inline src/pages/components/timeline.mdx:151]: re: `separator` I'd like to steer away from styling-related descriptions, if possible. Also, isn't the `icon` part of the rendered `Timeline.Item` (vs `Timeline.Content`)?

File: src/pages/components/stepper.mdx [hunk 2]
```diff
@@ -56,28 +56,61 @@ import IconsCode from '!!raw-loader!../../examples/accordions/stepper/StepperIco
...
+
+- Content for a horizontal stepper must be rendered externally. See [example](#layout) for details.
+
 import COMPONENTS from '../../examples/accordions/components.ts';
 
 ### Stepper
 
 <Component components={props.data.mdx.components} componentName="stepper" />
 
+The Stepper component provides state to its children.
```
[the remark this answers is not in the document]
Comment (smritimadan) [reply src/pages/components/stepper.mdx]: Not a recovered phrase, just added it to complete the picture about what each component what doing. But you are right, it doesn't provide too much value for the consumer. I'll remove it, we don't want noise. :-)

File: src/pages/components/stepper.mdx [hunk 3]
```diff
@@ -56,28 +56,61 @@ import IconsCode from '!!raw-loader!../../examples/accordions/stepper/StepperIco
...
+    <Stepper.Label />
+    <Stepper.Content />
+  </Stepper.Step>
+  {/* other steps */}
+</Stepper>
+```
+
+<!-- markdownlint-disable no-emphasis-as-heading -->
+
+**Implementation notes**
```
[the remark this answers is not in the document]
Comment (jzempel) [reply src/pages/components/stepper.mdx]: I like it.

**Review by jzempel — APPROVED** [review]

File: src/pages/components/accordion.mdx [hunk 4]
```diff
@@ -75,30 +75,58 @@ import AccordionSizeCode from '!!raw-loader!../../examples/accordions/accordion/
...
 
 <Component components={COMPONENTS} componentName="accordion.label" />
 
+Nest the Label within a [Header](#accordionheader) component.
+
 #### Accordion.Panel
 
 <Component components={COMPONENTS} componentName="accordion.panel" />
 
+The Panel component renders the section's content and provides the calculated styles for animations. Nest it
```
Comment (diana-oum) [inline src/pages/components/accordion.mdx:120]: I personally don't know what "calculated styles" means. Wondering if there's another way to say this more plainly?

File: src/pages/components/stepper.mdx [hunk 5]
```diff
@@ -56,6 +56,27 @@ import IconsCode from '!!raw-loader!../../examples/accordions/stepper/StepperIco
...
+</Stepper>
+```
+
+<!-- markdownlint-disable no-emphasis-as-heading -->
+
+**Stepper orientation**
+
+<!-- markdownlint-enable no-emphasis-as-heading -->
+
+- Content for a vertical stepper must be defined using the `Stepper.Content` for each step
```
Comment (diana-oum) [inline src/pages/components/stepper.mdx:77]: Another must! You can rearrange it like this: "Define the content for the vertical stepper using the 'Stepper.Content' for each step" The second bullet: "Render the content for a horizontal stepper externally. See example for details."

File: src/pages/components/stepper.mdx [hunk 6]
```diff
@@ -56,6 +56,27 @@ import IconsCode from '!!raw-loader!../../examples/accordions/stepper/StepperIco
...
+</Stepper>
+```
+
+<!-- markdownlint-disable no-emphasis-as-heading -->
+
+**Stepper orientation**
+
+<!-- markdownlint-enable no-emphasis-as-heading -->
+
+- Define the content for a vertical stepper using the `Stepper.Content` for each step
```
Comment (diana-oum) [inline src/pages/components/stepper.mdx:77]: Was there a period at the end of this originally?

**Review by diana-oum — APPROVED** [review]
Might be missing a period but all good to go! 🚢  it

**Pull request merged**